src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -232,9 +233,17 @@ int sd_dhcp_server_stop(sd_dhcp_server *server) {
         server->receive_message =
                 sd_event_source_unref(server->receive_message);
 
+        if (!server->bind_to_interface) {
```

Comment:
please drop this if check. we want destructors to be as robust as possible, hence if there's an object set here, destroy it. All our xyz_unref() functions become NOPs if you pass NULL to them, to make this easy and safe.